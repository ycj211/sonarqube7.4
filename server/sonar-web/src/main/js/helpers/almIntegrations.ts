/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { isLoggedIn } from './users';
import { CurrentUser, AlmOrganization } from '../app/types';

export function hasAdvancedALMIntegration(user: CurrentUser) {
  return (
    isLoggedIn(user) && (isBitbucket(user.externalProvider) || isGithub(user.externalProvider))
  );
}

export function isBitbucket(almKey?: string) {
  return almKey && almKey.startsWith('bitbucket');
}

export function isGithub(almKey?: string) {
  return almKey === 'github';
}

export function isVSTS(almKey?: string) {
  return almKey === 'microsoft';
}

export function isPersonal(organization?: AlmOrganization) {
  return Boolean(organization && organization.personal);
}

export function sanitizeAlmId(almKey?: string) {
  if (isBitbucket(almKey)) {
    return 'bitbucket';
  }
  return almKey;
}
