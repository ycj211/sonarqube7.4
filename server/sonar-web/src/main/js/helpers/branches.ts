/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { sortBy } from 'lodash';
import {
  BranchLike,
  Branch,
  BranchType,
  ShortLivingBranch,
  LongLivingBranch,
  PullRequest,
  MainBranch,
  BranchParameters
} from '../app/types';

export function isBranch(branchLike?: BranchLike): branchLike is Branch {
  return branchLike !== undefined && (branchLike as Branch).isMain !== undefined;
}

export function isShortLivingBranch(branchLike?: BranchLike): branchLike is ShortLivingBranch {
  return (
    isBranch(branchLike) &&
    !branchLike.isMain &&
    (branchLike as ShortLivingBranch).type === BranchType.SHORT
  );
}

export function isLongLivingBranch(branchLike?: BranchLike): branchLike is LongLivingBranch {
  return (
    isBranch(branchLike) &&
    !branchLike.isMain &&
    (branchLike as LongLivingBranch).type === BranchType.LONG
  );
}

export function isMainBranch(branchLike?: BranchLike): branchLike is MainBranch {
  return isBranch(branchLike) && branchLike.isMain;
}

export function isPullRequest(branchLike?: BranchLike): branchLike is PullRequest {
  return branchLike !== undefined && (branchLike as PullRequest).key !== undefined;
}

export function getPullRequestDisplayName(pullRequest: PullRequest) {
  return `${pullRequest.key} â€“ ${pullRequest.title}`;
}

export function getBranchLikeDisplayName(branchLike: BranchLike) {
  return isPullRequest(branchLike) ? getPullRequestDisplayName(branchLike) : branchLike.name;
}

export function getBranchLikeKey(branchLike: BranchLike) {
  return isPullRequest(branchLike) ? `pull-request-${branchLike.key}` : `branch-${branchLike.name}`;
}

export function getBranchQualityGateColor(status: string) {
  let indicatorColor = 'gray';
  if (status === 'ERROR') {
    indicatorColor = 'red';
  } else if (status === 'WARN') {
    indicatorColor = 'orange';
  } else if (status === 'OK') {
    indicatorColor = 'green';
  }
  return indicatorColor;
}

export function isSameBranchLike(a: BranchLike | undefined, b: BranchLike | undefined) {
  // main branches are always equal
  if (isMainBranch(a) && isMainBranch(b)) {
    return true;
  }

  // short- and long-living branches are compared by type and name
  if (
    (isLongLivingBranch(a) && isLongLivingBranch(b)) ||
    (isShortLivingBranch(a) && isShortLivingBranch(b))
  ) {
    return a.type === b.type && a.name === b.name;
  }

  // pull requests are compared by id
  if (isPullRequest(a) && isPullRequest(b)) {
    return a.key === b.key;
  }

  // finally if both parameters are `undefined`, consider them equal
  return a === b;
}

export function sortBranchesAsTree(branchLikes: BranchLike[]) {
  const result: BranchLike[] = [];

  const mainBranch = branchLikes.find(isMainBranch);
  const longLivingBranches = branchLikes.filter(isLongLivingBranch);
  const shortLivingBranches = branchLikes.filter(isShortLivingBranch);
  const pullRequests = branchLikes.filter(isPullRequest);

  // main branch is always first
  if (mainBranch) {
    result.push(
      mainBranch,
      ...getPullRequests(mainBranch.name),
      ...getNestedShortLivingBranches(mainBranch.name)
    );
  }

  // then all long-living branches
  sortBy(longLivingBranches, 'name').forEach(longLivingBranch => {
    result.push(
      longLivingBranch,
      ...getPullRequests(longLivingBranch.name),
      ...getNestedShortLivingBranches(longLivingBranch.name)
    );
  });

  // finally all orhpan pull requests and branches
  result.push(
    ...sortBy(pullRequests.filter(pr => pr.isOrphan), pullRequest => pullRequest.key),
    ...sortBy(shortLivingBranches.filter(branch => branch.isOrphan), branch => branch.name)
  );

  return result;

  /** Get all short-living branches (possibly nested) which should be merged to a given branch */
  function getNestedShortLivingBranches(mergeBranch: string) {
    const found: ShortLivingBranch[] = shortLivingBranches.filter(
      branch => branch.mergeBranch === mergeBranch
    );

    let i = 0;
    while (i < found.length) {
      const current = found[i];
      found.push(...shortLivingBranches.filter(branch => branch.mergeBranch === current.name));
      i++;
    }

    return sortBy(found, branch => branch.name);
  }

  function getPullRequests(base: string) {
    return sortBy(pullRequests.filter(pr => pr.base === base), pullRequest => pullRequest.key);
  }
}

export function getBranchLikeQuery(branchLike?: BranchLike): BranchParameters {
  if (isShortLivingBranch(branchLike) || isLongLivingBranch(branchLike)) {
    return { branch: branchLike.name };
  } else if (isPullRequest(branchLike)) {
    return { pullRequest: branchLike.key };
  } else {
    return {};
  }
}

// Create branch object from branch name or pull request key
export function fillBranchLike(
  branch?: string,
  pullRequest?: string
): ShortLivingBranch | PullRequest | undefined {
  if (branch) {
    return {
      isMain: false,
      mergeBranch: '',
      name: branch,
      type: BranchType.SHORT
    } as ShortLivingBranch;
  } else if (pullRequest) {
    return { base: '', branch: '', key: pullRequest, title: '' } as PullRequest;
  }
  return undefined;
}
