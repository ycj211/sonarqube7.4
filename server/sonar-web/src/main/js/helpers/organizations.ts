/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { isLoggedIn } from './users';
import { Organization, OrganizationSubscription, CurrentUser } from '../app/types';

export function isPaidOrganization(organization: Organization | undefined): boolean {
  return Boolean(organization && organization.subscription === OrganizationSubscription.Paid);
}

export function hasPrivateAccess(
  currentUser: CurrentUser,
  organization: Organization | undefined,
  userOrganizations: Organization[]
): boolean {
  return (
    !isPaidOrganization(organization) ||
    isCurrentUserMemberOf(currentUser, organization, userOrganizations)
  );
}

export function isCurrentUserMemberOf(
  currentUser: CurrentUser,
  organization: Organization | undefined,
  userOrganizations: Organization[]
): boolean {
  return Boolean(
    organization &&
      isLoggedIn(currentUser) &&
      ((organization.actions && organization.actions.admin) ||
        userOrganizations.some(org => org.key === organization.key))
  );
}
