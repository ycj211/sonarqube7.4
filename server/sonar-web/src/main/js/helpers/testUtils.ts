/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow, ShallowRendererProps, ShallowWrapper, ReactWrapper } from 'enzyme';
import { InjectedRouter } from 'react-router';
import { IntlProvider } from 'react-intl';

export const mockEvent = {
  target: { blur() {} },
  currentTarget: { blur() {} },
  preventDefault() {},
  stopPropagation() {}
};

export function click(element: ShallowWrapper | ReactWrapper, event = {}): void {
  // `type()` returns a component constructor for a composite element and string for DOM nodes
  if (typeof element.type() === 'function') {
    element.prop<Function>('onClick')();
    // TODO find out if `root` is a public api
    // https://github.com/airbnb/enzyme/blob/master/packages/enzyme/src/ReactWrapper.js#L109
    (element as any).root().update();
  } else {
    element.simulate('click', { ...mockEvent, ...event });
  }
}

export function clickOutside(event = {}): void {
  const dispatchedEvent = new MouseEvent('click', event);
  window.dispatchEvent(dispatchedEvent);
}

export function submit(element: ShallowWrapper | ReactWrapper): void {
  element.simulate('submit', {
    preventDefault() {}
  });
}

export function change(element: ShallowWrapper | ReactWrapper, value: string, event = {}): void {
  // `type()` returns a component constructor for a composite element and string for DOM nodes
  if (typeof element.type() === 'function') {
    element.prop<Function>('onChange')(value);
    // TODO find out if `root` is a public api
    // https://github.com/airbnb/enzyme/blob/master/packages/enzyme/src/ReactWrapper.js#L109
    (element as any).root().update();
  } else {
    element.simulate('change', {
      target: { value },
      currentTarget: { value },
      ...event
    });
  }
}

export function keydown(keyCode: number): void {
  const event = new KeyboardEvent('keydown', { keyCode } as KeyboardEventInit);
  document.dispatchEvent(event);
}

export function elementKeydown(element: ShallowWrapper, keyCode: number): void {
  const event = {
    currentTarget: { element },
    keyCode,
    preventDefault() {}
  };

  if (typeof element.type() === 'string') {
    // `type()` is string for native dom elements
    element.simulate('keydown', event);
  } else {
    element.prop<Function>('onKeyDown')(event);
  }
}

export function resizeWindowTo(width?: number, height?: number) {
  // `document.body.clientWidth/clientHeight` are getters by default, so we need to redefine them
  // pass `configurable: true` to allow to redefine the properties multiple times
  if (width) {
    Object.defineProperty(document.body, 'clientWidth', { configurable: true, value: width });
  }
  if (height) {
    Object.defineProperty(document.body, 'clientHeight', { configurable: true, value: height });
  }

  const resizeEvent = new Event('resize');
  window.dispatchEvent(resizeEvent);
}

export function doAsync(fn?: Function): Promise<void> {
  return new Promise(resolve => {
    setImmediate(() => {
      if (fn) {
        fn();
      }
      resolve();
    });
  });
}

// Create the IntlProvider to retrieve context for wrapping around.
const intlProvider = new IntlProvider({ locale: 'en' }, {});
const { intl } = intlProvider.getChildContext();
export function shallowWithIntl<C extends React.Component>(
  node: React.ReactElement<any>,
  options: ShallowRendererProps = {}
) {
  return shallow<C>(node, { ...options, context: { intl, ...options.context } });
}

export async function waitAndUpdate(wrapper: ShallowWrapper<any, any> | ReactWrapper<any, any>) {
  await new Promise(setImmediate);
  wrapper.update();
}

export function mockRouter(overrides: { push?: Function; replace?: Function } = {}) {
  return {
    createHref: jest.fn(),
    createPath: jest.fn(),
    go: jest.fn(),
    goBack: jest.fn(),
    goForward: jest.fn(),
    isActive: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    setRouteLeaveHook: jest.fn(),
    ...overrides
  } as InjectedRouter;
}
