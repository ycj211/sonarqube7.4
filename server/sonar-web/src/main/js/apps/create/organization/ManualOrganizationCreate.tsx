/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import OrganizationDetailsForm from './OrganizationDetailsForm';
import OrganizationDetailsStep from './OrganizationDetailsStep';
import PlanStep from './PlanStep';
import { formatPrice } from './utils';
import { OrganizationBase, Organization, SubscriptionPlan } from '../../../app/types';
import { translate } from '../../../helpers/l10n';

interface Props {
  createOrganization: (organization: OrganizationBase) => Promise<Organization>;
  className?: string;
  deleteOrganization: (key: string) => Promise<void>;
  onOrgCreated: (organization: string) => void;
  onlyPaid?: boolean;
  subscriptionPlans?: SubscriptionPlan[];
}

enum Step {
  OrganizationDetails,
  Plan
}

interface State {
  organization?: Organization;
  step: Step;
}

export default class ManualOrganizationCreate extends React.PureComponent<Props, State> {
  mounted = false;
  state: State = { step: Step.OrganizationDetails };

  componentDidMount() {
    this.mounted = true;
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  handleOrganizationDetailsStepOpen = () => {
    this.setState({ step: Step.OrganizationDetails });
  };

  handleOrganizationDetailsFinish = (organization: Required<OrganizationBase>) => {
    this.setState({ organization, step: Step.Plan });
    return Promise.resolve();
  };

  handlePaidPlanChoose = () => {
    if (this.state.organization) {
      this.props.onOrgCreated(this.state.organization.key);
    }
  };

  handleFreePlanChoose = () => {
    return this.createOrganization().then(key => {
      this.props.onOrgCreated(key);
    });
  };

  createOrganization = () => {
    const { organization } = this.state;
    if (organization) {
      return this.props
        .createOrganization({
          avatar: organization.avatar,
          description: organization.description,
          key: organization.key,
          name: organization.name || organization.key,
          url: organization.url
        })
        .then(({ key }) => key);
    } else {
      return Promise.reject();
    }
  };

  deleteOrganization = () => {
    const { organization } = this.state;
    if (organization) {
      this.props.deleteOrganization(organization.key).catch(() => {});
    }
  };

  render() {
    const { className, subscriptionPlans } = this.props;
    const startedPrice = subscriptionPlans && subscriptionPlans[0] && subscriptionPlans[0].price;
    const formattedPrice = formatPrice(startedPrice);

    return (
      <div className={className}>
        <OrganizationDetailsStep
          finished={this.state.organization !== undefined}
          onOpen={this.handleOrganizationDetailsStepOpen}
          open={this.state.step === Step.OrganizationDetails}
          organization={this.state.organization}>
          <OrganizationDetailsForm
            onContinue={this.handleOrganizationDetailsFinish}
            organization={this.state.organization}
            submitText={translate('continue')}
          />
        </OrganizationDetailsStep>

        {subscriptionPlans !== undefined && (
          <PlanStep
            createOrganization={this.createOrganization}
            deleteOrganization={this.deleteOrganization}
            onFreePlanChoose={this.handleFreePlanChoose}
            onPaidPlanChoose={this.handlePaidPlanChoose}
            onlyPaid={this.props.onlyPaid}
            open={this.state.step === Step.Plan}
            startingPrice={formattedPrice}
            subscriptionPlans={subscriptionPlans}
          />
        )}
      </div>
    );
  }
}
