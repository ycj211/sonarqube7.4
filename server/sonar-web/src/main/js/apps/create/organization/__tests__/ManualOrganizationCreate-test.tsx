/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import ManualOrganizationCreate from '../ManualOrganizationCreate';
import { waitAndUpdate } from '../../../../helpers/testUtils';

const organization = {
  avatar: 'http://example.com/avatar',
  description: 'description-foo',
  key: 'key-foo',
  name: 'name-foo',
  url: 'http://example.com/foo'
};

it('should render and create organization', async () => {
  const createOrganization = jest.fn().mockResolvedValue({ key: 'foo' });
  const onOrgCreated = jest.fn();
  const wrapper = shallowRender({ createOrganization, onOrgCreated });

  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();

  wrapper.find('OrganizationDetailsForm').prop<Function>('onContinue')(organization);
  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();

  wrapper.find('PlanStep').prop<Function>('onFreePlanChoose')();
  await waitAndUpdate(wrapper);
  expect(createOrganization).toBeCalledWith(organization);
  expect(onOrgCreated).toBeCalledWith('foo');
});

it('should preselect paid plan', async () => {
  const wrapper = shallowRender({ onlyPaid: true });

  await waitAndUpdate(wrapper);
  wrapper.find('OrganizationDetailsForm').prop<Function>('onContinue')(organization);
  await waitAndUpdate(wrapper);
  expect(wrapper.find('PlanStep').prop('onlyPaid')).toBe(true);
});

it('should roll back after upgrade failure', async () => {
  const createOrganization = jest.fn().mockResolvedValue({ key: 'foo' });
  const deleteOrganization = jest.fn().mockResolvedValue(undefined);
  const wrapper = shallowRender({ createOrganization, deleteOrganization });
  await waitAndUpdate(wrapper);

  wrapper.find('OrganizationDetailsForm').prop<Function>('onContinue')(organization);
  await waitAndUpdate(wrapper);

  wrapper.find('PlanStep').prop<Function>('createOrganization')();
  expect(createOrganization).toBeCalledWith(organization);

  wrapper.find('PlanStep').prop<Function>('deleteOrganization')();
  expect(deleteOrganization).toBeCalledWith(organization.key);
});

function shallowRender(props: Partial<ManualOrganizationCreate['props']> = {}) {
  return shallow(
    <ManualOrganizationCreate
      createOrganization={jest.fn()}
      deleteOrganization={jest.fn()}
      onOrgCreated={jest.fn()}
      subscriptionPlans={[{ maxNcloc: 100000, price: 10 }, { maxNcloc: 250000, price: 75 }]}
      {...props}
    />
  );
}
