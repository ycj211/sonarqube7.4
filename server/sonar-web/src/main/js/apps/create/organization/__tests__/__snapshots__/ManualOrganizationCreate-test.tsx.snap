// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and create organization 1`] = `
<div>
  <OrganizationDetailsStep
    finished={false}
    onOpen={[Function]}
    open={true}
  >
    <OrganizationDetailsForm
      onContinue={[Function]}
      submitText="continue"
    />
  </OrganizationDetailsStep>
  <PlanStep
    createOrganization={[Function]}
    deleteOrganization={[Function]}
    onFreePlanChoose={[Function]}
    onPaidPlanChoose={[Function]}
    open={false}
    startingPrice="billing.price_format.10"
    subscriptionPlans={
      Array [
        Object {
          "maxNcloc": 100000,
          "price": 10,
        },
        Object {
          "maxNcloc": 250000,
          "price": 75,
        },
      ]
    }
  />
</div>
`;

exports[`should render and create organization 2`] = `
<div>
  <OrganizationDetailsStep
    finished={true}
    onOpen={[Function]}
    open={false}
    organization={
      Object {
        "avatar": "http://example.com/avatar",
        "description": "description-foo",
        "key": "key-foo",
        "name": "name-foo",
        "url": "http://example.com/foo",
      }
    }
  >
    <OrganizationDetailsForm
      onContinue={[Function]}
      organization={
        Object {
          "avatar": "http://example.com/avatar",
          "description": "description-foo",
          "key": "key-foo",
          "name": "name-foo",
          "url": "http://example.com/foo",
        }
      }
      submitText="continue"
    />
  </OrganizationDetailsStep>
  <PlanStep
    createOrganization={[Function]}
    deleteOrganization={[Function]}
    onFreePlanChoose={[Function]}
    onPaidPlanChoose={[Function]}
    open={true}
    startingPrice="billing.price_format.10"
    subscriptionPlans={
      Array [
        Object {
          "maxNcloc": 100000,
          "price": 10,
        },
        Object {
          "maxNcloc": 250000,
          "price": 75,
        },
      ]
    }
  />
</div>
`;
